metric-schema-java/src/main/java/com/palantir/metric/schema/JavaGeneratorArgs.java

```diff
@@ -40,9 +48,24 @@
     /** Output directory for generated java code. */
     abstract Path output();
 
+    /** Used to add the libraryName tag to all metrics. */
+    abstract Optional<String> libraryName();
+
     /** The default Java package name for generated classes. */
     abstract String defaultPackageName();
 
+    @Value.Check
+    final void check() {
+        libraryName().ifPresent(value -> {
+            Preconditions.checkArgument(
```

Comment:
I think these restrictions are another example of why the libraryName should be separate from the root project name

Reply:
I don't quite understand - can you flesh our your concern more?

Reply:
Sorry I should have said more words. Its a bit surprising that by applying a plugin on some random sub project I am effectively putting constraints on  the root project. Usually plugin application only impacts the current project and its sub projects

Reply:
If someone fails this validation maybe they will be a little surprised, but (a) I have yet to come across a repo that would fail this validation and (b) it's a default anyway, so they could choose to override it anyway. 

I'm just trying to automatically do the right thing in the 99% of cases, the 1% of special snowflakes can still override it.